benchmarking/mod/bblock/vdbench.py

```diff
@@ -97,13 +97,54 @@ def prepare_run(self):
         self.cleanup()
 	# format
 	self.format_run()
-
+	#Add new method to check vdbench
+	def check_vdbench_pgrep(self, nodes, vdbench_node_num = 1, check_type="jobnum"):
+        user =  self.cluster["user"]
+        stdout, stderr = common.pdsh(user, nodes, "pgrep -x vdbench", option = "check_return")
+        res = common.format_pdsh_return(stdout)
+        if res != []:
+            vdbench_running_job_num = 0
+            vdbench_running_node_num = 0
+            for node in res:
+                vdbench_running_node_num += 1
+                vdbench_running_job_num += len(str(res[node]).strip().split('\n'))
+            if (check_type == "jobnum" and vdbench_running_job_num >= vdbench_node_num) or (check_type == "nodenum" and vdbench_running_node_num >= vdbench_node_num):
+                common.printout("WARNING","%d vdbench job still runing" % vdbench_running_job_num)
+                return True
+            else:
+                if check_type == "nodenum":
+                    common.printout("WARNING","Expecting %d nodes run vdbench, detect %d node runing" % (vdbench_node_num, vdbench_running_node_num))
+                if check_type == "jobnum":
+                    common.printout("WARNING","Expecting %d nodes run vdbench, detect %d node runing" % (vdbench_node_num, vdbench_running_job_num))
+                return False
+        common.printout("WARNING","Detect no vdbench job runing" % (vdbench_node_num, vdbench_running_node_num))
+        return False
+   #Updated wait_workload_to_stop and stop_workload
     def wait_workload_to_stop(self):
-        pass
-
+        common.printout("LOG","Waiting Workload to complete its work")
+        nodes = []
+        for client in self.benchmark["distribution"]:
+            nodes.extend(self.benchmark["distribution"][client])
+        max_check_times = 30
+        cur_check = 0
+        while self.check_vdbench_pgrep(nodes):
+            if cur_check > max_check_times:
+                break
+            time.sleep(10)
+            cur_check += 1
+        common.printout("LOG","Workload completed")
     def stop_workload(self):
-        pass
-
+        user = self.cluster["user"]
+        for client in self.benchmark["distribution"]:
+            nodes = self.benchmark["distribution"][client]
```

Comment:
This is checking in each node, not client, so use nodes = self.cluster["nodes_distribution"]